castle/cms/browser/discussion.py

```diff
@@ -0,0 +1,60 @@
+from Acquisition import aq_inner
+from Acquisition import aq_parent
+from plone.app.dexterity.behaviors.discussion import IAllowDiscussion
+from plone.app.discussion.browser.conversation import ConversationView
+from plone.app.discussion.interfaces import IDiscussionSettings
+from plone.registry.interfaces import IRegistry
+from Products.CMFCore.utils import getToolByName
+from zope.component import queryUtility
+
+
+IAllowDiscussion['allow_discussion'].description = (u'Allow discussion for this content '
+                                                    u'object. If set on folder, it will '
+                                                    u'enable for all items inside folder.')
+
+
+class CastleConversationView(ConversationView):
+
+    def enabled(self):
+        """
+        We only support dexterity anyways so do complete check here...
+
+        Returns True if discussion is enabled for this conversation.
+
+        This method checks five different settings in order to figure out if
+        discussion is enable on a specific content object:
+
+        1) Check if discussion is enabled globally in the plone.app.discussion
+           registry/control panel.
+
+        2) Check if the allow_discussion boolean flag on the content object is
+           set. If it is set to True or False, return the value. If it set to
+           None, try further.
```

Comment:
I'm going to leave it the way it is right now because any further customizing of the behavior starts getting sketchy with how plone.app.discucssion works.